### PR TITLE
[oree-845] Manifest support for new editor extension

### PR DESCRIPTION
**Pull request by nimal-outreach**
This PR brings a new extension type - Editor Extension.

The type of extension is **shell-editor**.
Regions property allows optional constraining of the editor host-context  (e.g. extension is available only on a Tasks page)

Contextual properties available: User, Org, and Client.

Optional settings - the recommended size of the iframe

### CODE REVIEW

File: src/manifest/extensions/editor/EditorExtension.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+   * @return {*}  {boolean}
+   * @memberof EditorExtension
+   */
+  init(context: OutreachContext): boolean {
+    let modified = false;
+
+    try {
+      // 1. copy url search parameters to context urlParams
+      const url = new URL(this.host.url!);
+      const searchParams = new URLSearchParams(url.search);
```
Comment (palootcenas-outreach) [inline src/manifest/extensions/editor/EditorExtension.ts:79]: I believe `url.search` is already an instance of `URLSearchParams` so this step is not needed.

File: src/manifest/extensions/editor/EditorExtension.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+      if (!Array.isArray(this.context)) {
+        issues.push('Context section is not an array. Value: ' + this.context);
+      }
+
+      this.context.forEach((context) => {
+        if (
+          !Object.values(UserContextKeys).includes(context as UserContextKeys) &&
+          !Object.values(ClientContextKeys).includes(context as ClientContextKeys)
+        ) {
+          issues.push('Context key is not one of the valid values for the application tab extension. Key: ' + context);
```
Comment (palootcenas-outreach) [inline src/manifest/extensions/editor/EditorExtension.ts:179]:
```suggestion
          issues.push('Context key is not one of the valid values for the application editor extension. Key: ' + context);
```

File: src/manifest/extensions/editor/EditorRegion.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  /**
+   * Sequence editor context host
+   */
+  SEQUENCE = 'sequenceEditor',
+
+  /**
+   * Other context host
+   */
+  OTHER = 'other',
```
Comment (palootcenas-outreach) [inline src/manifest/extensions/editor/EditorRegion.ts:37]: Why do we need `OTHER` ?

**Review by palootcenas-outreach — APPROVED** [review]

File: src/manifest/extensions/ExtensionGuards.ts [hunk 4]
```diff
@@ -227,3 +228,15 @@ export const isActionShellExtension = (extension: any): extension is KnowledgeSh
 
   return true;
 };
+
+export const isEditorShellExtension = (extension: any): extension is EditorExtension => {
```
Comment (sheff146) [inline src/manifest/extensions/ExtensionGuards.ts:232]: Can't this all be simplified to a one liner?
`return Object.prototype.hasOwnProperty.call(extension, 'type') && extension.type === ShellExtensionType.EDITOR;`
or even 
`return extension.type === ShellExtensionType.EDITOR;`

**Review by sheff146 — APPROVED** [review]

File: src/manifest/extensions/editor/EditorRegion.ts (hunk 3 above)
Comment (nimal-outreach) [reply src/manifest/extensions/editor/EditorRegion.ts:37]: It is Galaxy enum values used by the existing text editors - not sure in which exactly place of app is used

**Pull request merged**